### PR TITLE
Update Rnafusion for release v3.0.1

### PR DESCRIPTION
**Pull request by fevac**
## Description
- https://github.com/Clinical-Genomics/MTP-RNAFUSION/issues/31

### Added
- Fusion VCF file to deliverables
- Gene counts file to deliverables
- CRAM index files to deliverables
- Scout upload of a Rnafusion alignment CRAM file
- Swedac logo in delivery report

### Changed
- Rnafusion bundle filenames
- Replaced deprecated metrics:
  - 5_3_bias by median_5prime_to_3prime_bias
  - reads_aligned by reads_pairs_examined
- Updated default parameters
- Skip path validation in model 

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b add_rnafusion_vcf -a
    ```

#### Small case
- [x] `cg workflow rnafusion start <case-id>`
- [x] `cg workflow rnafusion store <case-id>` 
- [x] for downsampled cases: `cg generate delivery-report --force <case-id>` 
- [x] `cg upload -c <case-id>`

Expected test outcome
- [x] Params file is written with new default parameters
- [x] Case starts and ends successfully
- [x] Fusion VCF is stored in HK
- [x] Fusion VCF is delivered to Caesar
- [x] Fusion VCF uploaded to Scout
- [x] Gene counts file is stored in HK
- [x] Gene counts file is delivered to Caesar
- [x] CRAM index file is stored in HK
- [x] CRAM index file is delivered to Caesar
- [x] CRAM file uploaded to Scout
- [x] Delivery report includes median 5-3 bias
- [x] Delivery report includes Swedac logo

#### Case with no fusions
- [x] `cg workflow rnafusion start <case-id>`
- [x] `cg workflow rnafusion store <case-id>` 
- [x] for downsampled cases: `cg generate delivery-report --force <case-id>` 
- [x] `cg upload -c <case-id>`

Expected test outcome
- [x] Case starts and ends successfully
- [x] Case is successfully stored
- [x] Report can be generated
- [x] Case can be uploaded and Scout doesn't have optional fields

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/resources/rnafusion_bundle_filenames.csv [hunk 1]
```diff
@@ -70,6 +76,18 @@
   path_index: ~
   step: software-versions
   tag: software-versions
+- format: vcf
+  id: CASEID
+  path: PATHTOCASE/vcf_collect/CASEID_fusion_data.vcf
```
Comment (fevac) [inline cg/resources/rnafusion_bundle_filenames.csv:81]:
```suggestion
  path: PATHTOCASE/vcf/CASEID_fusion_data.vcf.gz
```

File: cg/meta/upload/scout/scout_config_builder.py [hunk 2]
```diff
@@ -66,10 +66,9 @@ def add_common_sample_files(
         case_sample: CaseSample,
     ) -> None:
         """Add common sample files for different analysis types."""
-        sample_id: str = case_sample.sample.internal_id
```
Comment (fevac) [inline cg/meta/upload/scout/scout_config_builder.py:69]: this has to be the internal case id

Comment (ivadym) [reply cg/meta/upload/scout/scout_config_builder.py:69]: This method is common to all pipelines, so we need to log the sample ID

File: cg/constants/delivery.py [hunk 3]
```diff
@@ -153,20 +153,21 @@
...
-    {"multiqc-json"},
     {"delivery-report"},
+    {"vcf-fusion"},
+    {"gene-counts"},
 ]
 
-RNAFUSION_ANALYSIS_SAMPLE_TAGS: list[set[str]] = []
+RNAFUSION_ANALYSIS_SAMPLE_TAGS: list[set[str]] = [
+    {"cram"},
+    {"cram-index"},
```
Comment (ChrOertlin) [inline cg/constants/delivery.py:169]: I have seen these tags also in MIP. Maybe we can start a
`class AnalysisSampleTags(StrEnum):`
?

File: cg/meta/report/report_api.py [hunk 4]
```diff
@@ -103,8 +103,18 @@ def get_delivery_report_from_hk(self, case_id: str, version: Version) -> str | N
...
-        """Return the file path of the uploaded to Scout file given its tag."""
-        raise NotImplementedError
+        """Return file path of the uploaded to Scout file given its tag."""
+        version: Version = self.housekeeper_api.last_version(bundle=case_id)
+        tags: list = self.get_hk_scout_file_tags(scout_tag=scout_tag)
+        uploaded_file: File = self.housekeeper_api.get_latest_file(
+            bundle=case_id, tags=tags, version=version.id
+        )
+        if not tags or not uploaded_file:
+            LOG.warning(
```
Comment (ChrOertlin) [inline cg/meta/report/report_api.py:113]: should this only be a warning or raise an error?

File: cg/meta/report/rnafusion.py [hunk 5]
```diff
@@ -47,15 +48,16 @@ def get_sample_metadata(
...
             duplicates=sample_metrics.pct_duplication,
             gc_content=sample_metrics.after_filtering_gc_content,
             input_amount=input_amount,
             insert_size=None,
             insert_size_peak=None,
-            mapped_reads=sample_metrics.reads_aligned
-            * 2
-            / sample_metrics.before_filtering_total_reads,
+            mapped_reads=get_mapped_reads_fraction(
+                mapped_reads=sample_metrics.read_pairs_examined * 2,
```
Comment (ChrOertlin) [inline cg/meta/report/rnafusion.py:58]: maybe we can make `mapped_reads` a property on `sample_metrics` that does the caluclation and you can call `sample_metrics.mapped_reads` here instead. Would make the code more readable and understandable.

File: cg/meta/report/rnafusion.py [hunk 6]
```diff
@@ -77,12 +79,8 @@ def get_genome_build(self, analysis_metadata: AnalysisModel) -> str:
     def get_report_accreditation(
```
Comment (ChrOertlin) [inline cg/meta/report/rnafusion.py:79]:
```suggestion
    def is_accredited(
```
would be more appropriate for this function. Perhaps out of scope, ignore if you want

File: cg/meta/workflow/nf_analysis.py [hunk 7]
```diff
@@ -269,12 +271,14 @@ def get_deliverables_template_content() -> list[dict]:
...
         """Return PipelineDeliverables for a given case."""
         deliverable_template: list[dict] = self.get_deliverables_template_content()
+        sample_id: str = self.status_db.get_samples_by_case_id(case_id).pop().internal_id
         files: list[FileDeliverable] = []
         for file in deliverable_template:
             for deliverable_field, deliverable_value in file.items():
                 if deliverable_value is None:
                     continue
                 file[deliverable_field] = file[deliverable_field].replace("CASEID", case_id)
+                file[deliverable_field] = file[deliverable_field].replace("SAMPLEID", sample_id)
```
Comment (ChrOertlin) [inline cg/meta/workflow/nf_analysis.py:281]: DeliverableField constants perhaps?

**Review by ChrOertlin — APPROVED** [review]
looks good! 👍

File: cg/constants/delivery.py (hunk 3 above)
Comment (fevac) [reply cg/constants/delivery.py:169]: This is a good point. Let's do it in a different PR. Opened issue here https://github.com/Clinical-Genomics/cg/issues/2743

File: cg/meta/report/report_api.py (hunk 4 above)
Comment (fevac) [reply cg/meta/report/report_api.py:113]: It was as warning so we don't want to change the behaviour here

File: cg/meta/report/rnafusion.py (hunk 6 above)
Comment (fevac) [reply cg/meta/report/rnafusion.py:79]: This is out of the scope. Leaving out

File: cg/meta/workflow/nf_analysis.py [hunk 8]
```diff
@@ -269,12 +271,14 @@ def get_deliverables_template_content() -> list[dict]:
     def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
         """Return PipelineDeliverables for a given case."""
         deliverable_template: list[dict] = self.get_deliverables_template_content()
+        sample_id: str = self.status_db.get_samples_by_case_id(case_id).pop().internal_id
```
[the remark this answers is not in the document]
Comment (fevac) [reply cg/meta/workflow/nf_analysis.py:274]: this is a good point. Currently we only run this on rnafusion which always is one single sample. so this is not an issue there. Coming pipelines might differ, so I agree with this change. Will do in a different PR.

File: cg/meta/report/report_api.py (hunk 4 above)
Comment (ivadym) [reply cg/meta/report/report_api.py:113]: It should only be a warning.  The delivery report has its own validation of these fields and if mandatory it throws an error

File: cg/meta/report/rnafusion.py (hunk 6 above)
Comment (ivadym) [reply cg/meta/report/rnafusion.py:79]: I will add it in a PR I'm working now, thanks @ChrOertlin

**Comment by fevac** [comment]
- Deployed stage
image

- Deployed prod
image

**Pull request merged**